python/ngen_config_gen/src/ngen/config_gen/mappings.py

```diff
@@ -0,0 +1,250 @@
+"""
+ngen.config_gen.mappings provides Hydrofabric version specific name mappings to
+Hydrofabric 2.0 names. Mappings are split up by hook and hf version. For
+example, the namespace `LinkedData_2_2` provides mappings _from_ Hydrofabric
+2.2 Linked Data (`divide-attributes`) column names to Hydrofabric 2.0 Linked
+Data (`model_attributes`) names.
+
+Use ngen.config_gen.mappings to rename pandas DataFrame columns used as input to a
+DefaultHookProvider or other HookProviders. This retains compatibility with
+existing hook objects and avoids the need to update names referenced in
+hook implementations.
+
+See `examples/generate_conus_hf_2_2` for a concrete example.
+"""
+
+from __future__ import annotations
+
+import typing
+from types import MappingProxyType
+
+from ._metaclass_var_collection import MetaClassVarCollection, pub_cls_vars
+
+
+class LinkedData_2_2(metaclass=MetaClassVarCollection[typing.Mapping[str, str]]):
```

Comment:
While this solves a bespoke problem in the hook mechanics used in this package, I'm curious if this could be used to invert the mapping and provide 2.0 -> 2.2 mappings as well.

Also, is there a clear path to future hydrofabric name changes using this approach?

Reply:
I don't see why not. The hook providers don't care or know anything about the field names they are providing (the only exception is `divide_id`) they just come from the column names from a pandas dataframe. So, if you can create a dictionary that maps from 2.0 -> 2.2 names I don't see any reason why you couldn't do the opposite.

My thinking is we just provide the mappings for version X of the HF -> 2.0 for compatibility sake.